packages/contracts-bedrock/contracts/L2/CrossDomainOwnable3.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { Predeploys } from "../libraries/Predeploys.sol";
+import { L2CrossDomainMessenger } from "./L2CrossDomainMessenger.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title CrossDomainOwnable2
+ * @notice This contract extends the OpenZeppelin `Ownable` contract for L2 contracts to be owned
+ *         by contracts on L1. Note that this contract is meant to be used with systems that use
+ *         the CrossDomainMessenger system. It will not work if the OptimismPortal is used
+ *         directly.
+ */
+abstract contract CrossDomainOwnable3 is Ownable {
+    /**
+     * @notice If true, the contract uses the cross domain _checkOwner function override. If false
+     *         it uses the standard Ownable _checkOwner function.
+     */
+    bool internal isLocal = true;
+
+    /**
+     * @notice Overrides the implementation of the `onlyOwner` modifier to check that the unaliased
```

Comment:
Ok so 2 things here
- Definitely prefer to not use `super._checkOwner()` in the `isLocal` case because of the revert message. If you could port the code internal to that and update the revert message to be crossdomainownable3: prefixed?
- Preference for an event in transfer ownership that emits `isLocal`. The set of events should be able to be used to recreate the current state of the contract. Right now its impossible to know anything about `isLocal`. I have a preference for defining a new event in this contract that emits the previous owner, new owner and is local. Not super opinionated on leaving the call is `super.transferOwnership`, we can leave it or replace it with an impl that sets the owner

Reply:
sounds good, will add the event. Already changed the `super._checkOwner()` for `require(owner() == _msgSender(), "CrossDomainOwnable3: caller is not the owner");` like @maurelian suggested. Thanks for the review and all the help!